src/Bounds.cpp

```diff
@@ -754,9 +754,54 @@ class Bounds : public IRVisitor {
 
     void visit(const Let *op) {
         op->value.accept(this);
-        scope.push(op->name, Interval(min, max));
+        Expr min_val = min, max_val = max;
+
+        // We'll either substitute the values in directly, or pass
+        // them in as variables and add an outer let (to avoid
+        // combinatorial explosion).
+        Expr min_var, max_var;
+        string min_name = op->name + ".min";
+        string max_name = op->name + ".max";
+
+        if (min_val.defined()) {
+            if (is_const(min_val)) {
```

Comment:
I don't think there's any way for the min/max of a let value to be a var. The let value would have to be a var itself, which doesn't happen.